### ISSUE TITLE
2.17 breaks bundle using @rollup/plugin-commonjs

### ISSUE THREAD

**Issue by dstaley** [rollup/rollup#3642]
- Rollup Version: v2.17.0
- Operating System (or Browser): N/A
- Node Version (if applicable): v12.16.3
- Link to reproduction (IMPORTANT, read below): https://repl.it/repls/GlaringCheeryGlitch

### Expected Behavior

The generated bundle either includes `commonjsHelpers` or does not reference it.

### Actual Behavior

The generated bundle references a non-existing variable named `commonjsHelpers`. You can see the output on line 428 of `output.js` in the REPL. It's also produced below:

```js
var freeGlobal =
  typeof commonjsGlobal == "object" &&
  commonjsGlobal &&
  commonjsHelpers.commonjsGlobal.Object === Object &&
  commonjsGlobal;
```

In v2.16.1, the same line is as follows:

```js
var freeGlobal =
  typeof commonjsGlobal == "object" &&
  commonjsGlobal &&
  commonjsGlobal.Object === Object &&
  commonjsGlobal;
```

`commonjsHelpers` is only referenced once in the bundle, so I'm fairly sure it's not being included. Since this works with the latest version of `@rollup/plugin-commonjs` on v2.16.1 but not on v2.17.0, I'm unsure of whether this is an issue that needs to be solved in rollup or in the plugin, so please feel free to move this issue if everything is working as intended in rollup.

Also, just as a note, this prevents the entire bundle from working correctly, so it's preventing us from updating to v2.17.

For reference, [this](https://github.com/lodash/lodash/blob/898b378f069ecb6c92b7713529985ba78ff34d31/.internal/freeGlobal.js) is the line from lodash that's being messed up.

**Comment by lukastaegert** [comment]
I'm looking into this, here is a minimal reproduction without the CJS plugin or lodash:
https://rollupjs.org/repl/?version=2.17.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMGhlbHBlcnMlMjBmcm9tJTIwJy4lMkZoZWxwZXJzLmpzJyUzQiU1Q25jb25zb2xlLmxvZyhoZWxwZXJzLmdsb2JhbC5PYmplY3QlMjAlM0QlM0QlM0QlMjBPYmplY3QpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyaGVscGVycy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjB2YXIlMjBnbG9iYWwlMjAlM0QlMjAlN0IlN0QlM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

Note that it should actually look like this:
https://rollupjs.org/repl/?version=2.16.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMGhlbHBlcnMlMjBmcm9tJTIwJy4lMkZoZWxwZXJzLmpzJyUzQiU1Q25jb25zb2xlLmxvZyhoZWxwZXJzLmdsb2JhbC5PYmplY3QlMjAlM0QlM0QlM0QlMjBPYmplY3QpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyaGVscGVycy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjB2YXIlMjBnbG9iYWwlMjAlM0QlMjAlN0IlN0QlM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

Fix should be upcoming.

**Comment by lukastaegert** [comment]
Fix at #3643, sorry for the regression.
